repl/src/main/scala/ammonite/repl/Storage.scala

```diff
@@ -90,6 +133,63 @@ object Storage{
       }
     }
 
+    def classFilesListSave(pkg: String,
+                           wrapper: String,
+                           dataList: Seq[(String, String)],
+                           imports: Imports,
+                           tag: String,
+                           importTreesList: Seq[ImportTree]): Unit = {
+      val dir = pkg.replace("/", "$div") + "." + wrapper.replace("/", "$div") + "0"
+      val codeCacheDir = compileCacheDir/dir
+      if(!exists(codeCacheDir)) {
+        mkdir(codeCacheDir)
+        write(
+          codeCacheDir/classFilesOrder,
+          upickle.default.write((tag, dataList.reverse), indent = 4)
+        )
+        write(codeCacheDir/"imports.list", upickle.default.write(imports))
+        write(codeCacheDir/"importTrees.list", upickle.default.write(importTreesList))
```

Comment:
Store the importTree also and load it and pass each importTree to ResolveSingleImportHook, to load imports